### PR TITLE
Bjorncs/athenz bouncer filter

### PR DESCRIPTION
**Pull request by bjorncs**
FYI @tokle @freva 

The new filter utilizes the passthru function of the existing user authentication filter. It assumes that the user authentication filter is configured in the same chain as well.

### CODE REVIEW

**Comment by bjorncs** [comment]
@gjoranv PTAL

**Comment by bjorncs** [comment]
I'll squash the first and third commit if you agree with the changes.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/athenz/filter/UserAuthWithAthenzPrincipalFilter.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                break;
+        }
+    }
+
+    private UserAuthenticationResult fromHttpRequest(DiscFilterRequest request) {
+        if (!request.containsAttribute(userAuthenticationPassThruAttribute)) {
+            throw new IllegalStateException("User authentication filter passthru attribute missing");
+        }
+        Integer statusCode = (Integer) request.getAttribute(userAuthenticationPassThruAttribute);
+        for (UserAuthenticationResult result : UserAuthenticationResult.values()) {
```
Comment (gjoranv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/athenz/filter/UserAuthWithAthenzPrincipalFilter.java]: UAR.values().stream().filter(r -> r.statusCode == statusCode).findAny() ?

**Review by gjoranv — APPROVED** [review]
Please see comment.

**Pull request merged**